pkg/engine/availabilitysets.go

```diff
@@ -28,10 +28,6 @@ func CreateAvailabilitySet(cs *api.ContainerService, isManagedDisks bool) Availa
 			avSet.AvailabilitySetProperties = &compute.AvailabilitySetProperties{
 				PlatformUpdateDomainCount: to.Int32Ptr(3),
 			}
-			if cs.Properties.MasterProfile.PlatformFaultDomainCount != nil {
```

Comment:
So it looks like in fact we were basing the FD on the api model already, we just weren't exposing that as a user-viewable property. Does it make sense to explicitly pass in a `PlatformFaultDomainCount` value for Availability Zone-configured AvailabilitySet VMs? (I know that's a mouthful)

@ritazh @sozercan fyi

Reply:
Apparently yes, you can fix the FaultDomainCount of a AZ enabled VMSS.
default is 1, and means MaxSpread and is recommended. but it can also be fixed to _n_
So if the property is there, I figured its best to apply it. if you want the default, don't set it.
If we apply the property for non AZ but don't apply if for AZ enabled, then we differ in behavior from VMSS.
https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-manage-fault-domains

Reply:
@jackfrancis It seems that I didn't digest your mouthful properly. You did point out AvailabilitySet, my answer was documentation about VMSS. And you were right I think. 
The template does not include any AvailabilitySet when we set Zones, and so, there is no point setting the FDCount as it won't even show up anywhere in the arm template. 
see https://github.com/Azure/aks-engine/blob/master/pkg/engine/masterarmresources.go#L20-L23
So I'll revert this part and send a new PR, we can run BackCompat tests on the branch again and take it from there